boggle/Boggle_works.py

```diff
@@ -0,0 +1,152 @@
+
+from random import *
+from math import *
+from Gui import *
+import string
+
+a = range(1,17)
+letters = []
+words = []
+winners = []
+
+## extra is a function used to find all the blocks of a certain letter
+extra = lambda search_list, elem:[i for i, x in enumerate(search_list) if x ==elem]
+
+
+##a dictionary of the block connections
+connect = {0:[1,4,5],1:[0,2,4,5,6],2:[1,3,5,6,7],3:[2,6,7],4:[0,1,5,8,9],
+	5:[0,1,2,4,6,8,9,10],6:[1,2,3,5,7,9,10,11],7:[6,3,2,10,11],8:[4,5,9,12,13],
+	9:[8,10,4,5,6,12,13,14],10:[5,6,7,9,11,13,14,15],11:[10,6,7,14,15],
+	12:[8,9,13],13:[12,8,9,10,14],14:[13,9,10,11,15],15:[14,10,11]}
+
+class Boggle(Gui):
+    def __init__(self):
+		Gui.__init__(self)
+		self.setup()
+
+##the boggle gui setup, still need to fix the 'shake' cmd so that it resets the board		
+    def setup(self):
+		global a

```

Comment:
I would make "a" a member variable of the Boggle instance, rather than using a global.